script/testing/junit/src/TracefileTest.java

```diff
@@ -118,6 +128,27 @@ public static DynamicTest executeSelectQuery(int num, String cur_sql) {
         return cur;
     }
 
+    public static void checkEquals(List<String> res, List<String> queryResult) throws Exception {
+        double precision = 0.000001;
+        for(int i=0;i<res.size();i++){
+            if(res.get(i)==null){
+                if(queryResult.get(i)!=null){
+                    throw new Exception("Value null Mismatch");
+                }
+            }else if(res.get(i).equals("")){
+                if(!queryResult.get(i).equals("")){
+                    throw new Exception("Value '' Mismatch");
+                }
+            }else{
+                double one = Double.parseDouble(res.get(i));
+                double two = Double.parseDouble(queryResult.get(i));
+                if(Math.abs(one-two)>precision){
+                    throw new Exception("Expected " + queryResult + " but have " + res);
+                }
+            }
```

Comment:
Is this trying to parse all values to a double and then check whether they are equal? What if the type isn't a double?